### PR TITLE
[update] UDS Provisioning new flow

### PR DESCRIPTION
**Pull request by mhatrevi**

### CODE REVIEW

**Review by ekarabu — DISMISSED** [review]

File: rom/dev/README.md [hunk 1]
```diff
@@ -235,13 +235,31 @@ The following flows are conducted exclusively when the ROM is operating in ACTIV
 The following flows are conducted when the ROM is operating in the manufacturing mode, indicated by a value of `DEVICE_MANUFACTURING` (0x1) in the `CPTRA_SECURITY_STATE` register `device_lifecycle` bits.
 
 #### UDS Provisioning
```
Comment (jhand2) [inline rom/dev/README.md:237]: We should clarify that this flow is optional. An integration can still choose to generate UDS in it's own way.

Comment (bharatpillilli) [reply rom/dev/README.md:237]: **One correction:** An integration can still choose to generate UDS in it's own way **"only in passive mode"**

Comment (mhatrevi) [reply rom/dev/README.md:237]: Updated.

**Review by jhand2 — DISMISSED** [review]

File: rom/dev/README.md [hunk 2]
```diff
@@ -232,16 +232,34 @@ The initialization step involves a traditional startup script for microcontrolle
...
+1. On reset, the ROM checks if the `UDS_PROGRAM_REQ` bit in the `SS_DBG_MANUF_SERVICE_REG_REQ` register is set. If the bit is set, ROM initiates the UDS seed programming flow by setting the `UDS_PROGRAM_IN_PROGRESS` bit in the `SS_DBG_MANUF_SERVICE_REG_RSP` register.
 
-2. In this procedure, the ROM retrieves a 512-bit value from the iTRNG and writes it to the address specified by the `UDS_SEED_OFFSET` register, utilizing DMA hardware assistance.
+2. ROM then retrieves a 512-bit value from the iTRNG, the UDS Seed programming base address from the `SS_UDS_SEED_PROGRAMMING_BASE_ADDR_L` and `SS_UDS_SEED_PROGRAMMING_BASE_ADDR_H` registers and the Fuse Controller's base address from the `SS_OTP_FC_BASE_ADDR_L` and `SS_OTP_FC_BASE_ADDR_H` registers.
 
-3. Following the DMA operation, the ROM updates the `UDS_PROGRAM_REQ` bit in the `CPTRA_DBG_MANUF_SERVICE_RSP_REG` register to either `UDS_PROGRAM_SUCCESS` or `UDS_PROGRAM_FAIL`, indicating the outcome of the operation.
+3. ROM then retrieves the SS_OTP_FC_UDS_GRANULARITY from `SS_DBG_MANUF_SERVICE_REG_REQ` register Bit3 to learn if the fuse row is accessible with 32-bit or 64-bit granularity.
```
Comment (swenson) [inline rom/dev/README.md:242]:
```suggestion
3. ROM then retrieves the `SS_OTP_FC_UDS_GRANULARITY` from `SS_DBG_MANUF_SERVICE_REG_REQ` register Bit3 to learn if the fuse row is accessible with 32-bit or 64-bit granularity.
```

File: rom/dev/README.md [hunk 3]
```diff
@@ -232,16 +232,34 @@ The initialization step involves a traditional startup script for microcontrolle
...
 
-4. The manufacturing process then polls this bit and continues with the fuse burning flow as outlined by the fuse controller specifications and SOC-specific VR methodologies.
+4. ROM then performs the following steps until all the 512-bits of the UDS seed are programmed:
```
Comment (swenson) [inline rom/dev/README.md:244]:
```suggestion
4. ROM then performs the following steps until all the 512 bits of the UDS seed are programmed:
```

File: rom/dev/README.md [hunk 4]
```diff
@@ -232,16 +232,34 @@ The initialization step involves a traditional startup script for microcontrolle
...
+5. After completing the write operation, ROM triggers the partition  digest operation performing the following steps:
+    1. The ROM writes the lower 32 bits of the UDS Seed programming base address to the `DIRECT_ACCESS_ADDRESS` register.
+    2. The ROM triggers the digest calculation command by writing 0x4 to the `DIRECT_ACCESS_CMD` register.
+    3. The ROM continuously polls the Fuse Controller's `STATUS` register until the DAI state returns to idle.
+
+6. ROM updates the `UDS_PROGRAM_SUCCESS` or the `UDS_PROGRAM_FAIL` bit in the `SS_DBG_MANUF_SERVICE_REG_RSP` register to indicate the outcome of the operation.
+
+7. ROM then resets the `UDS_PROGRAM_IN_PROGRESS` bit in the `SS_DBG_MANUF_SERVICE_REG_RSP` register to indicate completion of the programming.
+
+8. The manufacturing process then polls this bit and continues with the fuse burning flow as outlined by the fuse controller specifications and SOC-specific VR methodologies.
```
Comment (swenson) [inline rom/dev/README.md:262]: The docs also suggest that locking is not complete until the next reset. Should we also explicitly require that the part is reset and the digest is verified?

File: rom/dev/README.md (hunk 2 above)
Comment (mhatrevi) [reply rom/dev/README.md:242]: Updated.

File: rom/dev/README.md (hunk 3 above)
Comment (mhatrevi) [reply rom/dev/README.md:244]: Updated.

File: rom/dev/README.md (hunk 4 above)
Comment (mhatrevi) [reply rom/dev/README.md:262]: Not sure if that fits in the README. I'll chat with Bharat and update (if required) this separately.

**Review by swenson — APPROVED** [review]
LGTM

**Pull request merged**